LeaderboardBackend/Models/Entities/AccountConfirmation.cs

```diff
@@ -0,0 +1,43 @@
+using System.ComponentModel.DataAnnotations;
+using NodaTime;
+
+namespace LeaderboardBackend.Models.Entities;
+
+/// <summary>
+///     Represents a user account confirmation.
+/// </summary>
+public class AccountConfirmation
+{
+    /// <summary>
+    ///     The unique identifier of the `AccountConfirmation`.<br/>
+    ///     Generated on creation.
+    /// </summary>
+    public Guid Id { get; set; }
+
+    /// <summary>
+    ///     The ID of the `User` the `AccountConfirmation` is a part of.
+    /// </summary>
+    [Required]
+    public Guid UserId { get; set; }
+
+    /// <summary>
+    ///     Relationship model for `UserId`.
+    /// </summary>
+    public User User { get; set; } = null!;
+
+    /// <summary>
+    ///     The time this `AccountConfirmation` is created.
+    /// </summary>
+    public Instant CreatedAt { get; set; } = SystemClock.Instance.GetCurrentInstant();
+
+    /// <summary>
+    ///     The time this `AccountConfirmation` is used.
+    /// </summary>
+    public Instant? UsedAt { get; set; }
+
+    /// <summary>
+    ///     The time this `AccountConfirmation` expires, after which the associated `User` must
+    ///     request for another confirmation email.
+    /// </summary>
+    public Instant ExpiresAt { get; set; } = SystemClock.Instance.GetCurrentInstant() + Duration.FromHours(1);
```

Comment:
Move this logic out of the entity definition.

Reply:
So I'll set these values on object creation instead? I figured that since these entities are only ever created by the one method in `AccountConfirmationService`, it'd be fine to have these definitions here.